### PR TITLE
Add minutiae detection

### PR DESCRIPTION
**Pull request by andrew-farries**
Add an implementation of minutia detection.

The input is a binarised, thinned image. The output is a list of minutia: their types (bifurcation, ridge end), positions and a list of list of points connecting them along ridges to the next minutia.

Checked by unit tests but we have not run this on an actual thinned binarised fingerprint image.

Best reviewed by looking at the overall diff.

### CODE REVIEW

File: Sources/Extraction/MinutiaeDetection/MinutiaeDetection.c [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+static MinutiaType GetMinutiaType(int numberOfNeighbors) {
+  switch ( numberOfNeighbors ) {
+    case 1 : return RidgeEnd;
+    case 3: return Bifurcation;
+    default: return None;
+  }
+}
+
+static List GetPointsWithXNeighbors(int X, BinaryMap * image, List minutiae)
```
Comment (fatboab) [inline Sources/Extraction/MinutiaeDetection/MinutiaeDetection.c]: Took me a while to work out what X meant, I initially thought it was to do with the width, might be better using N?
Also, do we need to return the list here, can we not pass in a pointer to the list and directly add stuff to it?

Comment (andrew-farries) [inline Sources/Extraction/MinutiaeDetection/MinutiaeDetection.c]: Done in 63e8baff8170745cf0f84b634b789f31998bef60

File: Sources/Extraction/MinutiaeDetection/MinutiaeDetection.c [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  Ridge * ridgeCopy = calloc(1, sizeof(*ridgeCopy));
+  *ridgeCopy = r;
+  return ridgeCopy;
+}
+
+static List TraceRidge(Point point, Point prev, BinaryMap * image, List outputPoints) {
+  List_AddData(&outputPoints, CopyPoint(prev));
+  while ( minutiaeLocations.data[point.x][point.y] == None ) {
+    List_AddData(&outputPoints, CopyPoint(point));
+    List neighbors = GetActiveNeighburs(point, image);
```
Comment (fatboab) [inline Sources/Extraction/MinutiaeDetection/MinutiaeDetection.c:113]: Are we leaking these neighbours as well as the other one you mentioned...?

Comment (andrew-farries) [inline Sources/Extraction/MinutiaeDetection/MinutiaeDetection.c:113]: yes

**Pull request merged**